### PR TITLE
[CPM Integration Tests] Adding support for hibernated shoots

### PR DESCRIPTION
**Pull request by krgostev**
**How to categorize this PR?**

/area control-plane-migration
/kind test

**What this PR does / why we need it**:
This PR skips the parts of the integration tests for control plane migration that are not applicable for shoots that are hibernated. This parts are:
1. Creating a client that connects to the shoot.
1. Creation / Deletion of `Secrets` and `ServiceAccount`.
1. `Nodes` comparison.
1. Deploying `Guest Book Application`.

The CPM IT will still check for successful migration and non - namespaced orphaned resources. It will also fetch the `Machines` that if everything is as expected should not exist.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: test/framework/shootmigrationtest.go [hunk 1]
```diff
@@ -165,7 +165,11 @@ func (t *ShootMigrationTest) PopulateBeforeMigrationComparisonElements(ctx conte
 	if err != nil {
 		return
 	}
-	t.ComparisonElementsBeforeMigration.NodeNames, err = t.GetNodeNames(ctx, t.ShootClient)
+	if !t.Shoot.Status.IsHibernated {
```
Comment (plkokanov) [inline test/framework/shootmigrationtest.go]: I'd put the check in t.GetNodeNames() to repetition in `PopulateAfterMigrationComparisonElements`

File: test/system/shoot_cp_migration/migrate_test.go [hunk 2]
```diff
@@ -202,28 +209,30 @@ func afterMigration(ctx context.Context, t *ShootMigrationTest, guestBookApp app
 	if err := t.PopulateAfterMigrationComparisonElements(ctx); err != nil {
 		return err
 	}
+	if !t.Shoot.Status.IsHibernated {
```
Comment (plkokanov) [inline test/system/shoot_cp_migration/migrate_test.go]: Nit: you can move the code that has to be executed in both cases (not hibernated and hibernated shoot) here, then just return nil if `Shoot.Status.IsHibernated` is true and then add the checks that require the shoot to not be hibernated.

**Review by plkokanov — CHANGES_REQUESTED** [review]
Thanks for the PR. I just have a couple of suggestions.

File: test/system/shoot_cp_migration/migrate_test.go [hunk 3]
```diff
@@ -171,6 +173,11 @@ func beforeMigration(ctx context.Context, t *ShootMigrationTest, guestBookApp *a
 		return err
 	}
 
+	t.GardenerFramework.Logger.Infof("Shoot.Status.IsHibernated = %t", t.Shoot.Status.IsHibernated)
+	if t.Shoot.Status.IsHibernated {
```
Comment (schrodit) [inline test/system/shoot_cp_migration/migrate_test.go]: Nit: 
The log does not harm but I think it is only needed if the shoot is hibernated. 
Otherwise the next `ginkgo.By("Creating test Secret and Service Account")` will be logged anyway.
```suggestion
	if t.Shoot.Status.IsHibernated {
	  t.GardenerFramework.Logger.Infof("Shoot is hibernated no further tests are executed")
```

**Review by schrodit — COMMENTED** [review]
Otherwise lgtm

**Review by schrodit — APPROVED** [review]

**Comment by krgostev** [comment]
Thank you for your comments! The suggestions that you have provided should be implemented now.

**Review by plkokanov — APPROVED** [review]
lgtm

File: test/framework/shootmigrationtest.go [hunk 4]
```diff
@@ -119,6 +119,10 @@ func (t *ShootMigrationTest) MigrateShoot(ctx context.Context) error {
 
 // GetNodeNames uses the shootClient to fetch all Node names from the Shoot
 func (t *ShootMigrationTest) GetNodeNames(ctx context.Context, shootClient kubernetes.Interface) (nodeNames []string, err error) {
+	if t.Shoot.Status.IsHibernated {
```
Comment (ialidzhikov) [inline test/framework/shootmigrationtest.go:122]: Don't we need to consider both spec and status? There might be the case when hibernation is triggered just now and then we will have `spec.hibernation.enabled=true` and `.status.isHibernated=false`.

```suggestion
	if gardencorev1beta1helper.HibernationIsEnabled(t.Shoot) || t.Shoot.Status.IsHibernated {
```

Comment (krgostev) [reply test/framework/shootmigrationtest.go:122]: Hmm, this case was actually never considered, as a possible scenario, or at least not for this PR. @plkokanov  do you think that we need to address this possibility?

Comment (plkokanov) [reply test/framework/shootmigrationtest.go:122]: @ialidzhikov I would leave this for a future PR as I'm currently running this scenario manually and checking any corner cases that arise. We haven't thought for any additional changes to the integration test that might be required.

Comment (ialidzhikov) [reply test/framework/shootmigrationtest.go:122]: Okay, no objections from my side, we can consider it in the future.
I just wanted to point out that most of the places in Gardener and in the extension library check both `spec.hibernation.enabled` and `status.isHibernated` when considering whether Shoot is hibernated or not.

**Review by ialidzhikov — APPROVED** [review]
/lgtm
